core/src/main/java/org/apache/carbondata/core/datastore/chunk/impl/DimensionRawColumnChunk.java

```diff
@@ -106,9 +106,13 @@ public DimensionColumnPage convertToDimColDataChunkWithOutCache(int index) {
       for (int i = 0; i < dataChunks.length; i++) {
         if (dataChunks[i] != null) {
           dataChunks[i].freeMemory();
+          dataChunks[i] = null;
         }
       }
     }
+    if (rawData != null) {
+      rawData = null;
+    }
```

Comment:
No need for null check as rawData is being set only to null. Better to implement freeMemory method in super class AbstractRawColumnChunk and set rawData to null in that and call super.freeMemory() from child classes.